### DISCUSSION TITLE
Different colors for Light and Dark theme from system

### DISCUSSION THREAD

**Discussion by LilithSilver** [AvaloniaCommunity/Material.Avalonia#377] [Q&A] [answered]
Currently, the theme transitions correctly if the theme is specified via XML without a specified `BaseTheme`: `<materialStyles:MaterialTheme PrimaryColor="Purple" SecondaryColor="Green"/>`. For example, when the user changes themes from Light to Dark in Windows settings, the theme transitions automatically.

I want to replicate that behavior, except I want to have different colors per dark and light themes. As an example, note how the primary color changes for a dark theme in the [material guidelines](https://m2.material.io/design/color/dark-theme.html#anatomy). Is there an official way of doing this, perhaps through resource overrides? If not, is there a workaround?

This is similar to #247, but the solution there only works for fully-bespoke in-app selection of themes. I want to customize the behavior based on the _system_ theme. (Although I'm not sure how the first provided solution on that answer works for light/dark themes, so maybe that's the answer.)

**Comment by LilithSilver** [comment, marked as answer]
Answered my own question. This is how to replicate the behavior of `MaterialTheme` using `MaterialThemeBase`.

```c#
    public override void OnFrameworkInitializationCompleted()
    {
        base.OnFrameworkInitializationCompleted();

        if (PlatformSettings is IPlatformSettings settings)
        {
            settings.ColorValuesChanged += PlatformSettings_ColorValuesChanged;
            SetupTheme(settings.GetColorValues());
        }
    }

    private void PlatformSettings_ColorValuesChanged(object? sender, PlatformColorValues colors)
    {
        SetupTheme(colors);
    }

    private void SetupTheme(PlatformColorValues colors)
    {
        if (colors.ThemeVariant == PlatformThemeVariant.Dark)
        {
            this.LocateMaterialTheme<MaterialThemeBase>().CurrentTheme = Theme.Create(
                Theme.Dark,
                myPrimaryColor,
                mySecondaryColor
            );
        }
        else
        {
            this.LocateMaterialTheme<MaterialThemeBase>().CurrentTheme = Theme.Create(
                Theme.Light,
                myPrimaryColor,
                mySecondaryColor
            );
        }
    }
```

**Reply by SKProCH** [reply]
Sorry, seems like i'm too late to the party. Yes, this is one of the solutions for this. Current code doesn't actually has different colors for different base themes, but isn't too difficulty add ones here.
